### ISSUE TITLE
[xenial] Implement nightly CI upgrade test from 14.04 to 16.04

### ISSUE THREAD

**Issue by eloquence** [freedomofpress/securedrop#3969]
A full 14.04 to 16.04 upgrade test is too long-running to perform with each PR, but it would be useful to have such a test run on a nightly basis. The main motivation is to catch any regressions caused by  baseline system configuration drift as a result of security and non-security package updates.

This will likely need to include logic for the backup → do-release-upgrade → restore process. Assuming we follow through on creating a `securedrop-admin` wrapper for this process, we should aim to re-use code instead of duplicating logic.

This is blocked on #3832 and on #3968, and should be considered a stretch goal for the Trusty->Xenial migration (#3204).

**Comment by redshiftzero** [comment]
trusty will EOL at the end of this month, we will not have time to do this prior to that date
